test/utils/config.go

```diff
@@ -187,3 +192,36 @@ func testChangePort_mixedGlobalApp(t *testing.T, snap, app, servicePort string)
 	})
 
 }
+
+func TestAutoStart(t *testing.T, snapName string, testAutoStart bool) {
+	t.Run("autostart", func(t *testing.T) {
+		if testAutoStart {
+			TestProcessAutostartGlobal(t, snapName)
+		}
+	})
+}
+
+func TestProcessAutostartGlobal(t *testing.T, snapName string) {
+	t.Run("set and unset autostart", func(t *testing.T) {
+		t.Cleanup(func() {
+			SnapUnset(t, snapName, "autostart")
+			SnapStop(t, snapName)
+		})
+
+		SnapStop(t, snapName)
+		require.False(t, SnapServicesEnabled(t, snapName))
+		require.False(t, SnapServicesActive(t, snapName))
+
+		SnapSet(t, snapName, "autostart", "true")
+		require.True(t, SnapServicesEnabled(t, snapName))
+		require.True(t, SnapServicesActive(t, snapName))
+
+		SnapUnset(t, snapName, "autostart")
+		require.True(t, SnapServicesEnabled(t, snapName))
+		require.True(t, SnapServicesActive(t, snapName))
```

Comment:
I don't remember exactly, but shouldn't unsetting autostart fall back to the default, i.e. service stopped and disabled?

Reply:
SHouldn't unsetting autostart have no effect on the service?
https://github.com/canonical/edgex-snap-hooks/blob/main/options/autostart_test.go#L37

Reply:
Right. Here is the code: https://github.com/canonical/edgex-snap-hooks/blob/9f5a8f96ce2847cb2ae88eabc1f521846d74593c/options/autostart.go#L59-L60